geograypher/utils/indexing.py

```diff
@@ -77,3 +78,69 @@ def ensure_float_labels(query_array, full_array=None) -> (np.ndarray, dict):
     # Cast to float, since that's expected
     output_query_array = query_array.astype(float)
     return output_query_array, IDs_to_label
+
+
+def inverse_map_interpolation(
+    ijmap: np.ndarray, downsample: int = 1, fill: int = -1
+) -> np.ndarray:
+    """
+    Inverts the type of map that can be fed to skimage.transform.warp.
+
+    The basic construction is the pixel position of these maps is the position
+    in the destination image, and the value of the map is the position in the
+    source image. Therefore if location [20, 30] has value [22.2, 28.4], it
+    means that the destination image pixel [20, 30] will be sampled from the
+    source image at pixel [22.2, 28.4] (the sampler can choose to snap to the
+    closest integer value or interpolate nearby pixels).
+
+    By inverting, we seek to reverse this map through interpolation. In the
+    example above, in the output we would now have the location [22, 28]
+    have the value [20, 30] (or slightly different because it might
+    interpolate with nearby values).
+
+    Arguments:
+        ijmap ((2, H, W) numpy array): a map of the structure discussed above
+        downsample (int): Inverting (particularly griddata) can be expensive
```

Comment:
Given this is a fairly smooth function I imagine you could get away with aggressive downsampling, like 10+ pixels.